vaadin-spring/src/main/java/com/vaadin/flow/spring/scopes/VaadinRouteScope.java

```diff
@@ -398,4 +434,16 @@ private static UI getUI() {
         return ui;
     }
 
+    private static UI findPreservingUI(UI ui) {
+        VaadinSession session = ui.getSession();
+        for (UI sessionUi : session.getUIs()) {
+            String windowName = getWindowName(ui);
```

Comment:
Should it be `String windowName = getWindowName(sessionUi);` instead?
Otherwise `windowName.equals(getWindowName(ui))` compares the same things.

Reply:
Good catch.
Leftover after refactoring.

Reply:
Tests pass even with `String windowName = getWindowName(ui);`. Is it possible to cover it with extra test, or would it take efforts/mocking?

Reply:
the test is added